src/renderer/components/configure-recording-set-page/index.tsx

```diff
@@ -1,112 +1,44 @@
+
+/*
+ Represents the page states controlled by configure-recording-set.
+ 'external' represents pages external from this system
+  - it's used to determine which transition should be used for the entering home page
+*/
+export type RecordingPageState = 'home' | 'external' | 'metadata';
+export type MetadataState = 'list-preview' | 'oto-ini' | 'dvcfg';
+
+/*
+Base page to handle configure-recording-set, dvcfg, oto.ini, and list-preview pages
+*/
 const ConfigureRecordingSetPage: FC<{
   onNext: MouseEventHandler<HTMLElement>;
   onBack: MouseEventHandler<HTMLElement>;
   onSetSelected: Consumer<RecordingSet>;
 }> = ({ onNext, onBack, onSetSelected }) => {
-  const { t } = useTranslation();
-  const { recordingProject } = useContext(RecordingProjectContext);
-  const [projectFile = DUMMY_PROJECT_FILE, setProjectFile] = useLocalStorage<ProjectFile>(
-    getLSKey('ConfigureRecordingSetPage', 'projectFile'),
-    DUMMY_PROJECT_FILE,
-  );
-  const [recordingSets = [], setRecordingSets] = useLocalStorage<RecordingSet[]>(
-    getLSKey('ConfigureRecordingSetPage', 'recordingSets'),
-    [],
-  );
-  const [showingDetails = false, setShowingDetails] = useLocalStorage(
-    getLSKey('ConfigureRecordingSetPage', 'showingDetails'),
-    false,
-  );
-  const [canWrite, setCanWrite] = useState(false);
-
-  useEffect(() => {
-    if (recordingProject) {
-      (async (): Promise<void> => {
-        const { rootPath } = recordingProject;
-        const configFilePath = join(rootPath, PROJECT_CONFIG_FILENAME);
-
-        const fileExistence = await checkFileExistence(configFilePath);
-        if (!fileExistence) {
-          setProjectFile(DUMMY_PROJECT_FILE);
-        } else if (fileExistence === 'folder') {
-          throw new Error(`Config file cannot be read because it is a directory: ${configFilePath}`);
-        } else {
-          // TODO Write a loader
-          const projectState = JSON.parse(await readFile(configFilePath)) as ProjectFile;
-          setProjectFile(projectState);
-          setRecordingSets(projectState.recordingSets);
-        }
-        setCanWrite(true);
-      })();
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [naiveSerialize(recordingProject), setProjectFile, setRecordingSets]);
-
-  useEffect(() => {
-    if (projectFile.recordingSets.length) {
-      // FIXME Need real loading
-      setRecordingSets(projectFile.recordingSets);
-    }
-  }, [projectFile, setRecordingSets]);
-
-  const [chosenKey = 'C', setChosenKey] = useLocalStorage<ScaleKey>(
-    getLSKey('ConfigureRecordingSetPage', 'chosenKey'),
-    'C',
-  );
-  const [chosenOctave = 3, setChosenOctave] = useLocalStorage<SupportedOctave>(
-    getLSKey('ConfigureRecordingSetPage', 'chosenOctave'),
-    3,
-  );
-  const [chosenName = '', setChosenName] = useLocalStorage(getLSKey('ConfigureRecordingSetPage', 'chosenName'), '');
+  const [recordingSetState, setRecordingSetState] = useState<RecordingPageState>('external');
```

Comment:
Should useLocalStorage always be used instead of useState? If not, when should it be used instead of useState?

Reply:
Almost always, as where the project stands right now. It's hard for you to test it because you don't develop on a mac, but it would be very obvious when it breaks. As a rule of thumb, any state variable that needs to be persisted to allow the component state to be restored when the page "reloads" (like clicking the refresh button on a normal webpage) needs to be persisted to the local storage.

That reminds me, you should be able to refresh the page using F5 key when the dev tool is open. Try refreshing the page and see if it breaks (or forgets everything you just did).